frame/contracts/proc-macro/src/lib.rs

```diff
@@ -606,6 +607,53 @@ fn expand_functions(def: &EnvDef, expand_blocks: bool, host_state: TokenStream2)
 		let is_stable = f.is_stable;
 		let not_deprecated = f.not_deprecated;
 
+		// wrapped host function body call with traces when target is`runtime::contracts::strace` and level is `trace`
+        // e.g ./target/release/substrate-contracts-node -l "fatal,runtime::contracts::strace=trace" --dev
+		//WIP(pg): replace comment above with link once the doc has been added to the readme
+		let wrapped_body_with_trace = {
+			let params_fmt_str = params.clone().filter_map(|arg| match arg {
+				syn::FnArg::Receiver(_) => None,
+				syn::FnArg::Typed(p) => {
+					match *p.pat.clone() {
+						syn::Pat::Ident(ref pat_ident) => Some(pat_ident.ident.to_string()),
+						_ => None,
+					}
+				},
+			})
+			.map(|s| format!("{s}: {{:?}}"))
+			.collect::<Vec<_>>()
+			.join(", ");
+
+			let trace_fmt_str = format!("{}::{}({}) = {{:?}}", module, name, params_fmt_str);
+			let trace_fmt_args = params.clone().filter_map(|arg| match arg {
+				syn::FnArg::Receiver(_) => None,
+				syn::FnArg::Typed(p) => {
+					match *p.pat.clone() {
+						syn::Pat::Ident(ref pat_ident) => Some(pat_ident.ident.clone()),
+						_ => None,
+					}
+				},
+			});
+
+			let debug_buffer_fmt_str = format!("{}\n", trace_fmt_str);
+			let debug_buffer_fmt_args = trace_fmt_args.clone();
+
+			quote! {
+				if ::log::log_enabled!(target: "runtime::contracts::strace", ::log::Level::Trace) {
+					let result = #body;
+					
+					::log::trace!(target: "runtime::contracts::strace", #trace_fmt_str, #( #trace_fmt_args, )* result);
```

Comment:
I think we print the debug buffer to the console after execution. So logging it here is probably redundant.

Reply:
correct this is happening here
https://github.com/paritytech/substrate/blob/a679efd39af64b810cf6d5bd4ff2f491528abed4/frame/contracts/src/exec.rs#L1013-L1017

Reply:
and thus for reference can be logged for console with the new code with 
```
❯ cargo run --release -- --dev -l "fatal,runtime::contracts::strace=trace,runtime::contracts=debug"
```